packages/gateway/src/gateway.js

```diff
@@ -25,55 +25,58 @@ export async function gatewayGet(request, env, ctx) {
   const reqUrl = new URL(request.url)
   const cid = getCidFromSubdomainUrl(reqUrl)
 
-  const gatewayReqs = env.ipfsGateways.map(async (url) => {
-    const ipfsUrl = new URL('ipfs', url)
-    const controller = new AbortController()
-    const startTs = Date.now()
-    const timer = setTimeout(() => controller.abort(), env.REQUEST_TIMEOUT)
+  const gatewayReqs = env.ipfsGateways.map((gwUrl) =>
+    _gatewayFetch(gwUrl, cid, {
+      pathname: reqUrl.pathname,
+      timeout: env.REQUEST_TIMEOUT,
+    })
+  )
 
-    let response
-    try {
-      response = await fetch(
-        `${ipfsUrl.toString()}/${cid}${reqUrl.pathname || ''}`,
-        { signal: controller.signal }
+  try {
+    /** @type {GatewayResponse} */
+    let winnerGwResponse = await pAny(gatewayReqs)
```

Comment:
Why not use the `options` object of `p-any` to filter out the ones that are not `ok`?

Reply:
I tried it at first given it would be so easier, but then I get to the point where I need to track counters for requests failed and I don't know what gateway urls failed because they will throw instead of returning the GatewayResponse object that I will be able to parse

Reply:
I was thinking about this yesterday, and using `p-settle` after should be enough to track the fail counters. Need to try it out tomorrow